sound/soc/sof/sof-audio.h

```diff
@@ -97,7 +97,8 @@ struct snd_sof_dai_config_data {
  * struct sof_ipc_pcm_ops - IPC-specific PCM ops
  * @hw_params: Function pointer for hw_params
  * @hw_free: Function pointer for hw_free
- * @trigger: Function pointer for trigger
+ * @post_trigger: Function pointer for host DMA post-trigger actions
+ * @pre_trigger: Function pointer for host DMA pre-trigger actions
```

Comment:
pre_trigger first, then post_trigger?

The comment should say in relation to **what** they are pre and post.

It is also odd that the host DMA is triggered via IPC on the DSP side, it is confusing to read it like this.
These are IPC specific  pre and post trigger callbacks around the platform trigger callback. If they start host DMA or not, it does not matter much, they do other things than that, right?

Reply:
My guess is that it is a bit of both.
IPC3 have two sequences: IPC first, platform second or platform first, IPC second
IPC4 on the other hand have a twist in some trigger: IPC1 first, platform second, IPC2 second ?

In any case a clear description on what these pre/post triggers are relating to could help and in the IPC specific code a further comment can put the pieces into context.